src/Features/Core/Portable/Completion/Providers/AbstractEnumCompletionProvider.cs

```diff
@@ -0,0 +1,131 @@
+        {
+            if (symbol.ContainingType != null && symbol.ContainingType.TypeKind == TypeKind.Enum)
+            {
+                lock (_cachedDisplayAndInsertionTextLock)
+                {
+                    if (!Equals(_cachedDisplayAndInsertionText.containingType, symbol.ContainingType) || _cachedDisplayAndInsertionText.context != context)
+                    {
+                        var displayFormat = SymbolDisplayFormat.MinimallyQualifiedFormat
+                            .WithMemberOptions(SymbolDisplayMemberOptions.IncludeContainingType)
+                            .WithLocalOptions(SymbolDisplayLocalOptions.None);
+                        var containingTypeText = symbol.ContainingType.ToMinimalDisplayString(context.SemanticModel, context.Position, displayFormat);
+                        _cachedDisplayAndInsertionText = (symbol.ContainingType, context, containingTypeText);
+                    }
+
+                    var text = $"{_cachedDisplayAndInsertionText.containingTypeText}.{symbol.Name}";
+                    return (text, "", text);
+                }
+            }
+
+            return GetDefaultDisplayAndSuffixAndInsertionText(symbol, context);
+        }
+
+        protected override CompletionItem CreateItem(CompletionContext completionContext, string displayText,
+            string displayTextSuffix, string insertionText, List<ISymbol> symbols, SyntaxContext context, bool preselect,
+            SupportedPlatformData supportedPlatformData)
+        {
+            var rules = GetCompletionItemRules(symbols);
+
+            rules = rules.WithMatchPriority(preselect ? MatchPriority.Preselect : MatchPriority.Default);
+
+            var sortAndFilterText = RemoveNamespaceFromDisplayText(displayText);
+
+            return SymbolCompletionItem.CreateWithSymbolId(
+                displayText: displayText,
+                displayTextSuffix: displayTextSuffix,
+                insertionText: insertionText,
+                filterText: sortAndFilterText,
+                symbols: symbols,
+                contextPosition: context.Position,
+                sortText: sortAndFilterText,
+                supportedPlatforms: supportedPlatformData,
+                rules: rules);
+        }
+
+        private static string RemoveNamespaceFromDisplayText(string displayText)
+        {
+            // Remove the namespace part of an enum value:
+            // Namespace.EnumType.EnumValue -> EnumType.EnumValue
+            var typeAndValueSeparator = displayText.LastIndexOf('.');
+            if (typeAndValueSeparator > 0)
+            {
+                var namespaceSeparator = displayText.LastIndexOf('.', typeAndValueSeparator - 1);
+                if (namespaceSeparator > 0)
+                {
+                    return displayText[(namespaceSeparator + 1)..];
+                }
+            }
+
+            return displayText;
```

Comment:
why are you doing this?  why was this information in teh display text in the first place?  if it was needed for the display text, why is it not appropriate to spit into the final user code?